scripts/partition_manager.py

```diff
@@ -24,60 +25,112 @@ def item_is_placed(d, item, after_or_before):
 
 def remove_irrelevant_requirements(reqs):
     # Remove items dependencies to partitions which are not present
-    [[remove_item_not_in_list(reqs[x]['placement'][before_after], reqs.keys())
-      for x in reqs.keys() if 'placement' in reqs[x] and type(reqs[x]['placement']) == dict
-      and before_after in reqs[x]['placement'].keys()]
-     for before_after in ['before', 'after']]
+    for x in reqs.keys():
+        if 'inside' in reqs[x].keys():
+            [remove_item_not_in_list(reqs[x]['inside'], reqs.keys())]
+            if not reqs[x]['inside']:
+                del reqs[x]['inside']
+        if 'inherit_size' in reqs[x].keys():
+            [remove_item_not_in_list(reqs[x]['inherit_size'], reqs.keys())]
+            if not reqs[x]['inherit_size']:
+                del reqs[x]['inherit_size']
+        for before_after in ['before', 'after']:
+            if 'placement' in reqs[x] and type(reqs[x]['placement']) == dict and before_after in reqs[x]['placement'].keys():
+                [remove_item_not_in_list(reqs[x]['placement'][before_after], reqs.keys())]
+                if not reqs[x]['placement'][before_after]:
+                    del reqs[x]['placement'][before_after]
+
+
+
+def outermost(reqs, elem):
+    comparison = 0
+    while 'inside' in elem.keys() and elem['inside'][0] in reqs.keys():
+        comparison += 1
+        elem = reqs[elem['inside'][0]]
+    return comparison
+
+
+def get_images_which_need_resolving(reqs, sub_partitions):
+    # Get candidates
+    unsorted = [x for x in reqs.keys() if 'placement' in reqs[x] and type(reqs[x]['placement']) == dict
+                 and ('before' in reqs[x]['placement'].keys() or 'after' in reqs[x]['placement'].keys())]
+
+    # Sort sub_partitions by whether they are inside other sub_partitions. Innermost first.
+    sorted_subs = sorted(sub_partitions.values(), key=lambda x: outermost(sub_partitions, x), reverse=True)
+
+    # Sort candidates by whether they are part of a sub_partitions.
+    # sub_partition parts come last in the result list so they are more likely
+    # to end up being placed next to each other, since they are inserted last.
```

Comment:
If you have two partitions placed after the same third partition, you can have two orders:

    foo:
        placement: {after: app}

    bar
        placement: {after: app}

Order 1:

    app foo bar

Order 2:

    app bar foo

Which order ends up as the result is dependent on which of `foo` and `bar` is processed before the other. The one that is processed last, ends up immediately after the app.

Since PM fails if partitions in a span are not consecutive, this sorting minimizes the risk that PM fails in these ambiguous situations.

Reply:
The semantics of after are such that it only cares about the first existing entry, right?

Reply:
Yes

Reply:
Any update on this? AFAICT the order should not matter given my explanation above